analyzers/tests/SonarAnalyzer.UnitTest/Rules/ParametersCorrectOrderTest.cs

```diff
@@ -28,15 +28,16 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class ParametersCorrectOrderTest
     {
+        private readonly VerifierBuilder builderCS = new VerifierBuilder<CS.ParametersCorrectOrder>();
```

Comment:
I see this line repeated in a lot of tests. WDYT about creating a generic base class that would contain this logic and the specific tests would just provide the analyzer type to them?

Reply:
I like the idea. It is a little challenge to get two rules (VB and C# variant) both in, but that should also be possible.

Reply:
I had the same idea (of course, with having two base classes as @Corniel pointed out), but @costin-zaharia-sonarsource was not happy with that design.
I will not address this in this PR. We can have a separate discussion about it.

Reply:
I think that having a base class for all the tests can be a major trap. It will be too easy to put things there and soon it will get out of control.

Reply:
We should agree on not putting there anything else :)

Reply:
> We should agree on not putting there anything else :)

It's to say this now and later "forget" the rule just because it's again very convenient in the short term. I think that adding this line with the constructor call is not that bad.